crates/stackable-versioned-macros/tests/snapshots/stackable_versioned_macros__snapshot_tests__k8s@conversion_tracking.rs.snap

```diff
@@ -0,0 +1,130 @@
+            baz: __sv_foospec.baz.into(),
+        }
+    }
+}
+#[automatically_derived]
+impl ::std::convert::From<v1beta1::FooSpec> for v1alpha1::FooSpec {
+    fn from(__sv_foospec: v1beta1::FooSpec) -> Self {
+        Self {
+            baz: __sv_foospec.baz.into(),
+        }
+    }
+}
+#[automatically_derived]
+pub(crate) mod v1beta1 {
+    use super::*;
+    #[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, CustomResource)]
+    #[kube(
+        group = "stackable.tech",
+        version = "v1beta1",
+        kind = "Foo",
+        status = FooStatus
+    )]
+    pub struct FooSpec {
+        pub bah: usize,
+        pub baz: bool,
+    }
+}
+#[automatically_derived]
+impl ::std::convert::From<v1beta1::FooSpec> for v1::FooSpec {
+    fn from(__sv_foospec: v1beta1::FooSpec) -> Self {
+        Self {
+            bar: __sv_foospec.bah.into(),
+            baz: __sv_foospec.baz.into(),
+        }
+    }
+}
+#[automatically_derived]
+impl ::std::convert::From<v1::FooSpec> for v1beta1::FooSpec {
+    fn from(__sv_foospec: v1::FooSpec) -> Self {
+        Self {
+            bah: __sv_foospec.bar.into(),
+            baz: __sv_foospec.baz.into(),
+        }
+    }
+}
+#[automatically_derived]
+pub(crate) mod v1 {
+    use super::*;
+    #[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, CustomResource)]
+    #[kube(group = "stackable.tech", version = "v1", kind = "Foo", status = FooStatus)]
+    pub struct FooSpec {
+        pub bar: usize,
+        pub baz: bool,
+    }
+}
+#[automatically_derived]
+pub(crate) enum Foo {
+    V1Alpha1,
+    V1Beta1,
+    V1,
+}
+#[automatically_derived]
+impl ::std::fmt::Display for Foo {
+    fn fmt(
+        &self,
+        f: &mut ::std::fmt::Formatter<'_>,
+    ) -> ::std::result::Result<(), ::std::fmt::Error> {
+        match self {
+            Self::V1Alpha1 => f.write_str("v1alpha1"),
+            Self::V1Beta1 => f.write_str("v1beta1"),
+            Self::V1 => f.write_str("v1"),
+        }
+    }
+}
+#[automatically_derived]
+impl Foo {
+    /// Generates a merged CRD containing all versions and marking `stored_apiversion` as stored.
+    pub fn merged_crd(
+        stored_apiversion: Self,
+    ) -> ::std::result::Result<
+        ::k8s_openapi::apiextensions_apiserver::pkg::apis::apiextensions::v1::CustomResourceDefinition,
+        ::kube::core::crd::MergeError,
+    > {
+        ::kube::core::crd::merge_crds(
+            vec![
+                < v1alpha1::Foo as ::kube::core::CustomResourceExt > ::crd(), <
+                v1beta1::Foo as ::kube::core::CustomResourceExt > ::crd(), < v1::Foo as
+                ::kube::core::CustomResourceExt > ::crd()
+            ],
+            &stored_apiversion.to_string(),
+        )
+    }
+}
+#[derive(
+    ::core::clone::Clone,
+    ::core::fmt::Debug,
+    ::serde::Deserialize,
+    ::serde::Serialize,
+    ::schemars::JsonSchema
+)]
+#[serde(rename_all = "camelCase")]
+pub struct FooStatus {
```

Comment:
WDYT of calling the generated struct something like `FooStatusWithChangedValues`?
Currently this will probably clash with a lot of existing status structs (e.g. `TrinoClusterStatus`) and it's hard to correctly name them something else everywhere.

Reply:
Sure, we can name them differently. I'm not a huge fan of `FooStatusWithChangedValues` but less collisions should be the focus for now. We can fine-tune the name down the line.

Reply:
Done in e38161b.